Add image download scripts, Fedora Atomic

junousi opened the pull request:
Add support for downloading images instead of building in DIB. Add support for flagging os_distro metadata.
Add Fedora Atomic download+upload script.

Commit:
Add image download script

Add basic scripts for downloading an image from a certain
location, then deploying and testing it with existing functions.
Add a distro specific script for Fedora Atomic.

Commit:
Add os_distro property

Add os_distro property to images. Set it specifically for Fedora
Atomic, and as generic "linux" for others.

Commit:
Set proper image name

Set a more compact image name already into the filename when
downloading the image. Override the image name variables after
image_constants.sh defaults have been loaded. Update the Floating
IP to the one currently reserved for imgbuild tenant.

rlaurika commented:
@junousi Can you give this pull request a human readable title?

rlaurika commented on scripts/image_create_fedora_atomic_dl.sh:
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash --login
+REL_VERSION=$(curl -Ls https://download.fedoraproject.org/pub/alt/atomic/stable/ |grep -Eo 'Fedora-Atomic-25-[0-9\.]*'|sort -r|head -1)
```
Could we make the base URLs here configurable?

rlaurika replied:
I mean the "https://download.fedoraproject.org/pub/alt/atomic/stable/" bit that is common to both REL_VERSION and DOWNLOAD_URL.

junousi replied:
Added in latest commit

rlaurika commented on scripts/image_create_fedora_atomic_dl.sh:
```diff
@@ -0,0 +1,7 @@
+DOWNLOAD_URL="https://download.fedoraproject.org/pub/alt/atomic/stable/$REL_VERSION/CloudImages/x86_64/images/$REL_VERSION.x86_64.qcow2"
+IMAGE_NAME="Fedora-Atomic-25"
+
+source $(dirname $0)/image_dl_ul.sh "$DOWNLOAD_URL"
```
I'm a bit confused about this image_dl_ul.sh script, since I can't see it in the scripts directory. Am I missing something?

junousi replied:
Just referring to an old dev time script, fixed with latest commit.

rlaurika commented on scripts/image_dl_test_deploy.sh:
```diff
@@ -0,0 +1,17 @@
+source $(dirname $0)/image_constants.sh
+source $(dirname $0)/image_functions.sh
+#HACK for k8s. Fix before adding more download+upload style images.
+OS_DISTRO_PROPERTY="fedora-atomic"
```
Perhaps set this in image_create_fedora_atomic_dl.sh instead?

rlaurika commented on scripts/image_constants.sh:
```diff
@@ -9,7 +9,8 @@ SECURITY_GROUPS="default"
 TEST_PING_COUNT="5"
 TEST_PING_INTERVAL="20"
 PYTHONIOENCODING="utf-8"
+OS_DISTRO_PROPERTY="linux"
```
Should this be a constant at all? How about setting it per image instead like IMAGE_NAME? Then it can be properly set for all images without workarounds.

junousi replied:
I think that before heading out to vacation I was simply trying to refrain from changing the other build scripts if possible. It would be the better approach for sure. Stay tuned.

Commit:
Use proper download script name

Refer to the image_dl_test_deploy.sh script instead of
image_dl_ul.sh used during development.

Commit:
Configurable host URL for download

Use configurable variable containing protocol://host/path to be
used for image download.

Commit:
Move Atomic specific variables

Move Fedora Atomic specific variables to an atomic-specific script
from the common download-test-deploy script where they were initially
placed as a workaround.

Set os_distro contents to an empty string for images other than Atomic.

Add cronjob for atomic.

junousi commented:
@rlaurika os_distro is now configured at the Atomic build script level. I left the glance image-create call as generic. IOW it always adds the os_distro property; for other images it's value is just set as empty. Please let me know if you'd like more granularity (for example not set the property at all for some images but do set it as something specific for others).

junousi commented:
Also, amended cronjob.

oscarkraemer commented on scripts/image_create_fedora_atomic_dl.sh:
```diff
@@ -0,0 +1,9 @@
+IMAGE_NAME="Fedora-Atomic-25"
+OS_DISTRO_PROPERTY="fedora-atomic"
+
+source $(dirname $0)/image_dl_test_deploy.sh "$DOWNLOAD_URL"
```
test_deploy? The name ins not fitting if this will be used in production.

junousi replied:
Context:

$ grep function\ image_ image_functions.sh 
function image_create() {
function image_download() {
function image_test() {
function image_deploy() {

junousi replied:
...where image_test() has carried that name since 2015-04-01. If renaming/refactoring/sanitization is required, I'd suggest that would be handled via another task.

oscarkraemer replied:
aaah, okay now it makes sense.

oscarkraemer commented on scripts/image_dl_test_deploy.sh:
```diff
@@ -0,0 +1,14 @@
+image_test
+image_deploy
+delete_old_image_versions
+#FIXME delete_old_image_versions doesn't catch .qcow2?
```
FIXME?

junousi replied:
Yeah, scripts are not the best place to track work. Comment deleted.

Commit:
Remove a #FIXME
